hadoop-hdds/common/src/main/java/org/apache/hadoop/ozone/upgrade/BasicUpgradeFinalizer.java

```diff
@@ -71,6 +69,11 @@ public synchronized StatusAndMessages preFinalize(String upgradeClientID,
       return FINALIZATION_IN_PROGRESS_MSG;
     case FINALIZATION_DONE:
     case ALREADY_FINALIZED:
+      if (versionManager.needsFinalization()) {
```

Comment:
Can you please explain how this can happen, and how can a user recover from this?

Reply:
it cant happen. just trying to catch if any future modification messes this up.